DOC-1329: RS CLI Utilites Reorg - rladmin upgrade command page (#1924)
RedisLabs/redislabs-docs: content/rs/references/cli-utilities/rladmin/upgrade.md

For more information, see [Upgrade an existing Redis Software Deployment]({{< relref "/rs/installing-upgrading/upgrading" >}}).

```sh
rladmin upgrade db { db:<id> | <name> }
                [ preserve_roles ]
                [ keep_redis_version ]
                [ discard_data ]
                [ force_discard ]
                [ parallel_shards_upgrade ]
                [ keep_crdt_protocol_version ]
                [ redis_version <version> ]
                [ force ]
                [ { latest_with_modules | and module module_name <mod_name> version <version> module_args <args_str> } ]
```

### Parameters

| Parameters                 | Type/Value               | Description                                                                                                            |
|----------------------------|--------------------------|------------------------------------------------------------------------------------------------------------------------|
| db                         | db:\<id\> <br />name     | Database to upgrade                                                                                                    |

> | and module | [upgrade module](#upgrade-module) command | Clause that allows upgrade of database and specified Redis module in a single step with only one restart (can be specified multiple times)  |

Reviewer: ```suggestion
| and module | [upgrade module](#upgrade-module) command | Clause that allows the upgrade of a database and a specified Redis module in a single step with only one restart (can be specified multiple times)  |
```

| discard_data               |                          | Indicates that data will not be saved after the upgrade                                                                |
| force                      |                          | Forces upgrade and skips warnings and confirmations                                                                    |

> | force_discard              |                          | Forces `discard_data` if there is replication or persistence enabled                                                   |

Reviewer: ```suggestion
| force_discard              |                          | Forces `discard_data` if replication or persistence is enabled                                                   |
```

> | keep_crdt_protocol_version |                          | Keeps the current crdt protocol version                                                                                |

Reviewer: ```suggestion
| keep_crdt_protocol_version |                          | Keeps the current CRDT protocol version                                                                                |
```

| keep_current_version       |                          | Upgrades to a new patch release, not to the latest major.minor version                                                 |

> | latest_with_modules        |                          | Upgrades both the database and all modules in the database                                                             |

Reviewer: ```suggestion
| latest_with_modules        |                          | Upgrades the Redis Enterprise Software version and all modules in the database                                                             |
```

| parallel_shards_upgrade    | Integer <br />all        | Maximum number of shards to upgrade all at once                                                                        |

> | preserve_roles             |                          | Performs an additional failover to guarantee roles of shards are preserved                                             |

Reviewer: ```suggestion
| preserve_roles             |                          | Performs an additional failover to guarantee the shards' roles are preserved                                             |
```

> | redis_version              | Redis version            | Upgrades database to the specified version instead of the latest version                                               |

Reviewer: ```suggestion
| redis_version              | Redis version            | Upgrades the database to the specified version instead of the latest version                                               |
```

$ rladmin upgrade db db:5
Monitoring e39c8e87-75f9-4891-8c86-78cf151b720b
active - SMUpgradeBDB init
active - SMUpgradeBDB check_slaves
.active - SMUpgradeBDB prepare
active - SMUpgradeBDB stop_forwarding
oactive - SMUpgradeBDB start_wd
active - SMUpgradeBDB wait_for_version
.completed - SMUpgradeBDB
Done
```

## `upgrade module`

Upgrades Redis modules in use by a specific database.

For more information, see [Upgrade modules]({{< relref "/modules/install/upgrade-module" >}}).

```sh
rladmin upgrade module
        db_name { db:<id> | <name> }
        module_name <mod_name>
        version <version>
        module_args <args_str>
```

### Parameters

| Parameters                 | Type/Value               | Description                                                                                                            |
|----------------------------|--------------------------|------------------------------------------------------------------------------------------------------------------------|
| db_name                    | db:\<id\> <br />name     | Database to with module to upgrade                                                                                     |

> | module_name                | Module name, such as: <br />ReJSON<br />graph<br />search<br />bf<br />timeseries | Redis module to upgrade                                       |

Reviewer: ```suggestion
| module_name                | 'ReJSON'<br />'graph'<br />'search'<br />'bf'<br />'timeseries' | Redis module to upgrade                                       |
```
